haskoin-wallet/Network/Haskoin/Wallet/Client/Commands.hs

```diff
@@ -154,26 +154,32 @@ getKey = do
         Just s -> go (cs s) derivM
         Nothing -> error "No action due to EOF"
   where
+    -- Default case. Generate a seed and derive account key using given or
+    -- default derivation.
     go "" derivM = return
         ( Nothing
         , Nothing
         , derivM <|> Just defaultDeriv
         , Nothing
         )
     go str' derivM = case xPrvImport str' of
+        -- Provided account private key. No further derivation should be
+        -- required unless one was given.
         Just k -> return
             ( Nothing
-            , Just $ maybe k (`derivePath` k) derivM
```

Comment:
Need to check if all the logic is still correct here. Instead of doing one odd derivation client side, all derivations are now computed server side with the given derivation path.